src/main/java/application/onlinebookstore/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package application.onlinebookstore.service.impl;
+
+import application.onlinebookstore.dto.cartitem.CartItemDto;
+import application.onlinebookstore.dto.cartitem.CartItemDtoUpdate;
+import application.onlinebookstore.dto.cartitem.CreateCartItemDto;
+import application.onlinebookstore.exception.EntityNotFoundException;
+import application.onlinebookstore.mapper.CartItemMapper;
+import application.onlinebookstore.model.CartItem;
+import application.onlinebookstore.model.ShoppingCart;
+import application.onlinebookstore.repository.book.BookRepository;
+import application.onlinebookstore.repository.cartitem.CartItemRepository;
+import application.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import application.onlinebookstore.service.CartItemService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CartItemServiceImpl implements CartItemService {
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public CartItemDto save(CreateCartItemDto cartItemDto, Long id) {
+        CartItem newCartItem = new CartItem();
+        newCartItem.setBook(bookRepository.findById(cartItemDto.bookId()).orElseThrow(

```

Comment:
Is it possible to make book service to be responsible for findById and handling optional?